File: console_print.js
#!/usr/bin/env node
/**
 * Simple JavaScript file for console printing examples.
 */

// Part 1: Direct code execution (no function)
console.log("=== Part 1: Direct Execution ===");
console.log("Hello from JavaScript!");
const message = "This is a simple print statement";
console.log(message);
console.log(`2 + 2 = ${2 + 2}`);
console.log();

// Part 2: Function definition and call
console.log("=== Part 2: Function Call ===");

/**
 * Function that prints a greeting message.
 * @param {string} name - The name to greet
 */
function greet(name) {
    console.log(`Hello, ${name}!`);
    console.log(`Welcome to JavaScript programming.`);
}

/**
 * Function that calculates and prints the sum of two numbers.
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} The sum of a and b
 */
function calculateSum(a, b) {
    const result = a + b;
    console.log(`The sum of ${a} and ${b} is ${result}`);
    return result;
}

// Calling the functions
greet("World");
calculateSum(10, 20);

File: console_print.py
#!/usr/bin/env python3
"""
Simple Python file for console printing examples.
"""

# Part 1: Direct code execution (no function)
print("=== Part 1: Direct Execution ===")
print("Hello from Python!")
message = "This is a simple print statement"
print(message)
print(f"2 + 2 = {2 + 2}")
print()

# Part 2: Function definition and call
print("=== Part 2: Function Call ===")


def greet(name):
    """Function that prints a greeting message."""
    print(f"Hello, {name}!")
    print(f"Welcome to Python programming.")


def calculate_sum(a, b):
    """Function that calculates and prints the sum of two numbers."""
    result = a + b
    print(f"The sum of {a} and {b} is {result}")
    return result


# Calling the functions
greet("World")
calculate_sum(10, 20)
